DefectDojo/django-DefectDojo
# replace django-tagging by django-tagulous

This PR fixes #1812, replacing `django-tagging` by `django-tagulous`. As well as #2993, #2569, #1849.

DONE tasks:
- create unit tests:
- test for finding:
		create with tags
		list and check tags
		get tags
		set tags
		/findings/{id}/tags/
			get, post (post adds to existing)
		/findings/{id}/remove_tags/			
			put, patch (both put and patch remove provided tags from existing, but throw error if tag was not already there)
		tests around spaces and comma's and quotes in tags
- test for import/reimport:
		set tags
- adjust prefetching
- update ui everywhere with new autocomplete field
- update finding bulk edit
- update apply template to finding
- add filtering on tags to UI (fixes #2993)
- update searching by tags (fixes #2569) 
- troubleshoot and report bug to tagulous around fixtures
- rename `Objects` model because it is a reserved keyword/attribute resulting problems with many-to-many relationships
- find out "how the #@$# can we get the width of the tags field correct"
- force lowercase (as was the case with django-tagging)
- update fixtures demodata, testdata, initial
- update serializers for REST API
- add filtering by tag to REST API (#1849)
- set autocomplete / initial for non modelforms ie forms with a tags field
- database migration
- work around `django-tagulous` not supporting serialization to `celery`
- use autocomplete field in filters
- support filtering on multiple tags in APIv2
- make notifications asynchronous again (had to be set to synchronous due to serialization problems)

FIXED some unrelated bugs along the way:
- jira edit engagement not working without JIRA
- add/edit tags for findings wasn't always working
- edit endpoint was not working for invalid data
- bring back in sync the reimport logic in the UI with APIv2

OBSERVATIONS (nothing changed)
- engagement_all page filters on products not on engagements
- bulk edit overwrites existing tags
